model/repository.js

```diff
@@ -143,3 +151,78 @@ module.exports.createOrUpdateRepository = function (name, data, callback) {
 module.exports.deleteRepositoryByName = function (name, callback) {
   Repository.findOneAndRemove({name: name}, callback);
 };
+
+/**
+ * Increment the build number for a registered repository
+ * @param name: `full_name` of the repository
+ * @param callback
+ */
+module.exports.incrementBuildNumber = function (name, callback) {
+  Repository.findOne({name: name}, 'builds', function (error, repository) {
+    if (error) {
+      callback(error);
+    } else {
+      repository.builds += 1;
+      repository.save(function (error, repository) {
+        callback(error, repository);
+      })
+    }
+  })
+};
+
+
+/**
+ * Get a single repository with a log history of 10 logs
+ * @param name: `full_name` of the repository
+ * @param callback
+ *//*
+module.exports.getRepositoryWithLogs = function (name, callback) {
+  Repository.getRepositoryByName(name, function (error, repository) {
+    if (error) {
+      callback(error);
+    } else {
+      BuildLog.getBuildLogsByRepository(name, function (error, buildLogs) {
+        callback(error, {
+          repository: repository,
+          buildLogs: buildLogs,
+        });
+      })
+    }
+  })
+}*/
+
+module.exports.getRepositoryWithLogs = function (name, callback) {
+  Repository.aggregate([
+    { $match: {name: name}},
+    {
+      $lookup: {
+        from: 'buildlogs',
+        localField: 'name',
+        foreignField: 'repository',
+        as: 'logs'
+      }
+    }
```

Comment:
use `$limit` https://docs.mongodb.com/manual/reference/operator/aggregation/limit/ . I think you can limit the number of log.